consensus-types/blocks/getters.go

```diff
@@ -0,0 +1,592 @@
+		return initBlindedSignedBlockFromProtoBellatrix(cp)
+	default:
+		return nil, errIncorrectBlockVersion
+	}
+}
+
+// PbGenericBlock returns a generic signed beacon block.
+func (b *SignedBeaconBlock) PbGenericBlock() (*eth.GenericSignedBeaconBlock, error) {
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	switch b.version {
+	case version.Phase0:
+		return &eth.GenericSignedBeaconBlock{
+			Block: &eth.GenericSignedBeaconBlock_Phase0{Phase0: pb.(*eth.SignedBeaconBlock)},
+		}, nil
+	case version.Altair:
+		return &eth.GenericSignedBeaconBlock{
+			Block: &eth.GenericSignedBeaconBlock_Altair{Altair: pb.(*eth.SignedBeaconBlockAltair)},
+		}, nil
+	case version.Bellatrix:
+		return &eth.GenericSignedBeaconBlock{
+			Block: &eth.GenericSignedBeaconBlock_Bellatrix{Bellatrix: pb.(*eth.SignedBeaconBlockBellatrix)},
+		}, nil
+	case version.BellatrixBlind:
+		return &eth.GenericSignedBeaconBlock{
+			Block: &eth.GenericSignedBeaconBlock_BlindedBellatrix{BlindedBellatrix: pb.(*eth.SignedBlindedBeaconBlockBellatrix)},
+		}, nil
+	default:
+		return nil, errIncorrectBlockVersion
+	}
+
+}
+
+// PbPhase0Block returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbPhase0Block() (*eth.SignedBeaconBlock, error) {
+	if b.version != version.Phase0 {
+		return nil, errNotSupported("PbPhase0Block", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBeaconBlock), nil
+}
+
+// PbAltairBlock returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbAltairBlock() (*eth.SignedBeaconBlockAltair, error) {
+	if b.version != version.Altair {
+		return nil, errNotSupported("PbAltairBlock", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBeaconBlockAltair), nil
+}
+
+// PbBellatrixBlock returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbBellatrixBlock() (*eth.SignedBeaconBlockBellatrix, error) {
+	if b.version != version.Bellatrix {
+		return nil, errNotSupported("PbBellatrixBlock", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBeaconBlockBellatrix), nil
+}
+
+// PbBlindedBellatrixBlock returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbBlindedBellatrixBlock() (*eth.SignedBlindedBeaconBlockBellatrix, error) {
+	if b.version != version.BellatrixBlind {
+		return nil, errNotSupported("PbBlindedBellatrixBlock", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBlindedBeaconBlockBellatrix), nil
+}
+
+// Version of the underlying protobuf object.
+func (b *SignedBeaconBlock) Version() int {
+	return b.version
+}
+
+// Header converts the underlying protobuf object from blinded block to header format.
+func (b *SignedBeaconBlock) Header() (*eth.SignedBeaconBlockHeader, error) {
+	if b.IsNil() || b.block.IsNil() {
```

Comment:
lets add one more check for the body, since we are doing a htr of it below.

Reply:
I don't know why I changed the implementation of `SignedBeaconBlock.IsNil`. I reverted it back to what's in the current block code, which basically checks everything for nilness: signed block, block and block body.